dart/lib/src/http_client/failed_request_client.dart

```diff
@@ -106,30 +112,50 @@ class FailedRequestClient extends BaseClient {
       rethrow;
     } finally {
       stopwatch.stop();
+      await _captureEventIfNeeded(
+        request,
+        statusCode,
+        exception,
+        stackTrace,
+        response,
+        stopwatch.elapsed,
+      );
+    }
+  }
 
-      // If captureFailedRequests is true, there statusCode is null.
-      // So just one of these blocks can be called.
-      var capture = false;
-      String? reason;
-      if (_hub.options.captureFailedRequests && exception != null) {
-        capture = true;
-      } else if (failedRequestStatusCodes.containsStatusCode(statusCode)) {
-        // Capture an exception if the status code is considered bad
-        capture = true;
-        reason = 'HTTP Client Error with status code: $statusCode';
-        exception ??= SentryHttpClientError(reason);
+  Future<void> _captureEventIfNeeded(
+      BaseRequest request,
+      int? statusCode,
+      Object? exception,
+      StackTrace? stackTrace,
+      StreamedResponse? response,
+      Duration duration) async {
+    if (!_hub.options.captureFailedRequests) {
+      return;
+    }
+
+    // Only check `failedRequestStatusCodes` & `failedRequestTargets` if no exception was thrown.
+    if (exception == null) {
+      if (!failedRequestStatusCodes.containsStatusCode(statusCode)) {
+        return;
       }
-      if (capture) {
-        await _captureEvent(
-          exception: exception,
-          stackTrace: stackTrace,
-          request: request,
-          requestDuration: stopwatch.elapsed,
-          response: response,
-          reason: reason,
-        );
+      if (!containsTracePropagationTarget(
```

Comment:
We can rename this function since its not only about `TracePropagation` anymore.